src/Analysis/Ast/Impl/Modules/PythonModule.cs

```diff
@@ -313,7 +313,16 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
             _linkedParseCts = CancellationTokenSource.CreateLinkedTokenSource(_disposeToken.CancellationToken, _parseCts.Token);
 
             ContentState = State.Parsing;
-            _parsingTask = Task.Run(() => Parse(_linkedParseCts.Token), _linkedParseCts.Token);
+            _parsingTask = Task.Run(() => ParseAndLogExceptions(_linkedParseCts.Token), _linkedParseCts.Token);
+        }
+
+        private void ParseAndLogExceptions(CancellationToken cancellationToken) {
+            try {
```

Comment:
another approach will be having something like this

```csharp
ErrorHandler.LogIfNotCancelled(Exception ex, ILogger logger, string message) {
	if (ex is Operation...) return false;
	logger?.Log($"exception : {ex}, message: {message}");
    return false;
}

try
{
}
catch(Exception ex) where (ErrorHandler.LogIfNotCancelled(ex, logger, filePath)) { }

so every one could use same helper.

Reply:
I'll pull that code out and see how it looks. Maybe just locally and we can pull it out all the way later.

Reply:
I'm going to skip it for now, but we can pull it out later if needed. I'm not seeing other places in the current code where it would help.